### DISCUSSION TITLE
Find package using cmake

### DISCUSSION THREAD

**Discussion by topazus** [xmake-io/xmake#5627] [Q&A]
The fmt-devel package is installed, but I tried to find fmt using cmake in xmake, but it does not find.

```bash
❯ xmake l find_package cmake::GTest
{
  links = {
    "gtest"
  },
  defines = {
    "GTEST_LINKED_AS_SHARED_LIBRARY"
  },
  libfiles = {
    "/usr/lib64/libgtest.so"
  },
  linkdirs = {
    "/usr/lib64"
  }
}

❯ xmake l find_package cmake::fmt
nil

❯ rpm -ql fmt-devel
/usr/include/fmt
/usr/include/fmt/args.h
/usr/include/fmt/base.h
/usr/include/fmt/chrono.h
/usr/include/fmt/color.h
/usr/include/fmt/compile.h
/usr/include/fmt/core.h
/usr/include/fmt/format-inl.h
/usr/include/fmt/format.h
/usr/include/fmt/os.h
/usr/include/fmt/ostream.h
/usr/include/fmt/printf.h
/usr/include/fmt/ranges.h
/usr/include/fmt/std.h
/usr/include/fmt/xchar.h
/usr/lib64/cmake/fmt
/usr/lib64/cmake/fmt/fmt-config-version.cmake
/usr/lib64/cmake/fmt/fmt-config.cmake
/usr/lib64/cmake/fmt/fmt-targets-relwithdebinfo.cmake
/usr/lib64/cmake/fmt/fmt-targets.cmake
/usr/lib64/libfmt.so
/usr/lib64/pkgconfig/fmt.pc
```

**Comment by star-hengxing** [comment]
Try `pkgconfig::fmt` or use offical package `add_requires("fmt")`

**Reply by topazus** [reply]
Yes, `pkgconfig::fmt` can be found, but why does `cmake::fmt` can not found?

**Comment by waruqi** [comment]
Using cmake to find packages, its implementation has some hacks. It's not stable and needs to rely on various cmake parameter configurations. I don't recommend using the cmake package in the presence of the xmake-repo package.

if you want to use it, you can debug here.

https://github.com/xmake-io/xmake/blob/dev/xmake/modules/package/manager/cmake/find_package.lua
